startrhacm.sh

```diff
@@ -188,158 +210,316 @@ if [[ "${FAILED}" == "true" ]]; then
-if [[ "${DOWNSTREAM}" == "true" ]]; then
-  printlog info "Getting downstream snapshot"
-  queryquay "acm-d"
+  printlog info "Updating Openshift pull-secret in namespace openshift-config with a token for quay.io:433"
+  QUAY443_TOKEN=$(echo "${QUAY_TOKEN}" | base64 --decode | sed "s/quay\.io/quay\.io:443/g")
+  OPENSHIFT_PULL_SECRET=$(oc get -n openshift-config secret pull-secret -o jsonpath='{.data.\.dockerconfigjson}' | base64 --decode)
+  FULL_TOKEN="${QUAY443_TOKEN}${OPENSHIFT_PULL_SECRET}"
+  oc set data secret/pull-secret -n openshift-config --from-literal=.dockerconfigjson="$(jq -s '.[0] * .[1]' <<<"${FULL_TOKEN}")"
 
-# If DOWNSTREAM is not "true", get snapshot from pipeline repo (defaults to latest edge version)
-else
-  if [[ -z "${RHACM_SNAPSHOT}" ]]; then
-    printlog info "Getting upstream snapshot"
-    cd ${RHACM_PIPELINE_PATH}
-    git pull &>/dev/null
-    BRANCH=${RHACM_BRANCH:-$(git remote show origin | grep -o " [0-8]\+\.[0-9]\+-" | sort -uV | tail -1 | grep -o "[0-9]\+\.[0-9]\+")}
-    VERSION_NUM=${RHACM_VERSION:=""}
-    PIPELINE_PHASE=${PIPELINE_PHASE:-"dev"}
-    # Handle older pipeline phases
-    if [[ "${BRANCH}" == "2."[0-4] ]]; then
-      case "${PIPELINE_PHASE}" in
-        dev|nightly)
-          PIPELINE_PHASE="edge"
-          ;;
-        preview)
-          PIPELINE_PHASE="stable"
-          ;;
-      esac
-    fi
-    printlog info "Updating repo and switching to the ${BRANCH}-${PIPELINE_PHASE} branch (if this exits, check the state of the local Pipeline repo)"
-    git checkout ${BRANCH}-${PIPELINE_PHASE} &>/dev/null
-    git pull &>/dev/null
-    if (! ls ${RHACM_PIPELINE_PATH}/snapshots/manifest-* &>/dev/null); then
-      printlog error "The branch, ${BRANCH}-${PIPELINE_PHASE}, doesn't appear to have any snapshots/manifest-* files to parse a snapshot from."
-      if [[ -z "${RHACM_BRANCH}" ]]; then
-        BRANCH=${RHACM_BRANCH:-$(git remote show origin | grep -o " [0-8]\+\.[0-9]\+-" | sort -uV | tail -2 | head -1 | grep -o "[0-9]\+\.[0-9]\+")}
-        printlog info "RHACM_BRANCH was not set. Using an older branch: ${BRANCH}-${PIPELINE_PHASE}"
-        git checkout ${BRANCH}-${PIPELINE_PHASE} &>/dev/null
-        git pull &>/dev/null
-      else
-        printlog error "Please double check the Pipeline repo and set RHACM_BRANCH as needed."
-        exit 1
-      fi
+  printlog info "Applying ImageDigestMirrorSet"
+  oc apply -f - <<EOF
+apiVersion: config.openshift.io/v1
+kind: ImageDigestMirrorSet
+metadata:
+  name: image-mirror-custom
+spec:
+  imageDigestMirrors:
+    - mirrors:
+        - 'quay.io:443/acm-d'
+        - brew.registry.redhat.io/rh-osbs/rhacm2
+      source: registry.redhat.io/rhacm2
+    - mirrors:
+        - 'quay.io:443/acm-d'
+        - brew.registry.redhat.io/rh-osbs/multicluster-engine
+      source: registry.redhat.io/multicluster-engine
+    - mirrors:
+        - 'quay.io:443/acm-d'
+      source: registry.redhat.io/openshift4
```

Comment:
nit: I'm not sure the quotes nor the brew registries are necessary?

```suggestion
  imageDigestMirrors:
    - mirrors:
        - quay.io:443/acm-d
      source: registry.redhat.io/rhacm2
    - mirrors:
        - quay.io:443/acm-d
      source: registry.redhat.io/multicluster-engine
    - mirrors:
        - quay.io:443/acm-d
      source: registry.redhat.io/openshift4
```

Reply:
Could this even just be this? I'm not so familiar with digest mirrors, so I'm not sure.

```yaml
  imageDigestMirrors:
    - mirrors:
        - quay.io:443/acm-d
      source: registry.redhat.io
```

Reply:
I'm not sure - I had already slimmed this down from an original doc I was working from :laughing:

I think your single mirror entry would probably work, but might mean that some images might unintentionally get pulled from quay.